patches/server/0891-Fire-EntityChangeBlockEvent-in-more-places.patch

```diff
@@ -110,6 +126,25 @@ index 68a8d3b16d49c10fc9834f32009095d35c9c55a8..1f8b7b50c6aa24778d87821ae2ff4d01
              if (player instanceof ServerPlayer) {
                  CriteriaTriggers.ITEM_USED_ON_BLOCK.trigger((ServerPlayer)player, blockPos, itemStack);
              }
+diff --git a/src/main/java/net/minecraft/world/item/PotionItem.java b/src/main/java/net/minecraft/world/item/PotionItem.java
+index 0b344e20ad4093b135c58905fb82ffff05898bb2..356cbdd1f015507b9c307f18d0e97fb5e2228360 100644
+--- a/src/main/java/net/minecraft/world/item/PotionItem.java
++++ b/src/main/java/net/minecraft/world/item/PotionItem.java
+@@ -107,6 +107,14 @@ public class PotionItem extends Item {
+         BlockState iblockdata = world.getBlockState(blockposition);
+ 
+         if (context.getClickedFace() != Direction.DOWN && iblockdata.is(BlockTags.CONVERTABLE_TO_MUD) && PotionUtils.getPotion(itemstack) == Potions.WATER) {
++            // Paper start
++            if (org.bukkit.craftbukkit.event.CraftEventFactory.callEntityChangeBlockEvent(entityhuman, blockposition, Blocks.MUD.defaultBlockState()).isCancelled()) {
++                if (!entityhuman.getAbilities().instabuild) {
++                    ((net.minecraft.server.level.ServerPlayer) entityhuman).getBukkitEntity().updateInventory();
++                }
```

Comment:
You have to update the inventory for all players, even those with the instabuild ability. Otherwise the player's in creative mode will get an empty bottle added to their inventory.

Reply:
oh yes i was just too focused on the item in hand :sweat_smile: